### PR TITLE
Add tee info hash tool

### PR DESCRIPTION
**Pull request by liuw1**
Support MRTD calculation
Add sample manifest
Generate tee info hash binary

Signed-off-by: Wei Liu <wei3.liu@intel.com>

### CODE REVIEW

File: td-shim-tools/src/tee_info_hash.rs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+use std::io::Seek;
+use std::io::SeekFrom;
+use std::mem::size_of;
+use td_layout::metadata::*;
+use zeroize::Zeroize;
+
+pub const SHA384_DIGEST_SIZE: usize = 48;
+
+const PAGE_SIZE: u64 = 0x1_000;
+const METADATA_PTR_OFFSET: usize = 0x01_000_000 - 0x20;
```
Comment (jyao1) [inline td-shim-tools/src/tee_info_hash.rs:22]: why 0x01_000_000 ?

Comment (liuw1) [reply td-shim-tools/src/tee_info_hash.rs:22]: Currently, td shim image size is 16M.

Comment (jyao1) [reply td-shim-tools/src/tee_info_hash.rs:22]: dont make assumption in tool, please

Comment (jyao1) [reply td-shim-tools/src/tee_info_hash.rs:22]: please use ImageSize - 0x20

Comment (liuw1) [reply td-shim-tools/src/tee_info_hash.rs]: Resolved

File: td-shim-tools/src/tee_info_hash.rs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+            xfam: Default::default(),
+            mrtd: [0; SHA384_DIGEST_SIZE],
+            mrconfig_id: [0; SHA384_DIGEST_SIZE],
+            mrowner: [0; SHA384_DIGEST_SIZE],
+            mrownerconfig: [0; SHA384_DIGEST_SIZE],
+            rtmr0: [0; SHA384_DIGEST_SIZE],
+            rtmr1: [0; SHA384_DIGEST_SIZE],
+            rtmr2: [0; SHA384_DIGEST_SIZE],
+            rtmr3: [0; SHA384_DIGEST_SIZE],
+            reserved: [0; 112],
```
Comment (jyao1) [inline td-shim-tools/src/tee_info_hash.rs:83]: please dont use hardcoded 112.
Define MACRO for it.

File: td-shim-tools/src/tee_info_hash.rs [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        buffer[packed_size..packed_size + SHA384_DIGEST_SIZE].copy_from_slice(&self.rtmr1);
+        packed_size += SHA384_DIGEST_SIZE;
+        buffer[packed_size..packed_size + SHA384_DIGEST_SIZE].copy_from_slice(&self.rtmr2);
+        packed_size += SHA384_DIGEST_SIZE;
+        buffer[packed_size..packed_size + SHA384_DIGEST_SIZE].copy_from_slice(&self.rtmr3);
+        packed_size += SHA384_DIGEST_SIZE;
+        // buffer[packed_size..packed_size + SHA384_DIGEST_SIZE].copy_from_slice(&self.servtd_hash);
+        // packed_size += SHA384_DIGEST_SIZE;
+        buffer[packed_size..packed_size + 112].copy_from_slice(&self.reserved);
+        packed_size += 112;
```
Comment (jyao1) [inline td-shim-tools/src/tee_info_hash.rs:138]: please dont use hardcoded 112.
Define MACRO for it.

File: td-shim-tools/src/tee_info_hash.rs [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+        packed_size
+    }
+
+    pub fn build_mrtd(&mut self, raw_image_file: &mut File, image_size: u64) {
+        raw_image_file
+            .seek(SeekFrom::Start(image_size - 0x20 as u64))
```
Comment (jyao1) [inline td-shim-tools/src/tee_info_hash.rs:144]: please dont use hardcoded 0x20.
Define MACRO for it.

**Comment by liuw1** [comment]
Resolved hardcode implementations.

File: td-shim-tools/src/tee_info_hash.rs [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    /// Software-defined ID for owner-defined configuration of
+    /// the guest TD
+    pub mrownerconfig: [u8; 48],
+    /// Runtime extendable measurement registers
+    pub rtmr0: [u8; 48],
+    pub rtmr1: [u8; 48],
+    pub rtmr2: [u8; 48],
+    pub rtmr3: [u8; 48],
+    /// Reserved. Must be zero
+    pub reserved: [u8; 112],
```
Comment (jyao1) [inline td-shim-tools/src/tee_info_hash.rs:73]: TD_INFO_STRUCT_RESERVED_SIZE ?

Comment (liuw1) [reply td-shim-tools/src/tee_info_hash.rs]: Just removed the unused traits pread and pwrite of the struct, then the struct can use MACRO.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**